src/internal/bind-to-namespace.ts

```diff
@@ -0,0 +1,21 @@
+/**
+ * Bind all methods of a class to the class instance.
+ *
+ * @param moduleClass The class to bind the methods of.
+ * @param that The class instance to bind the methods to.
+ */
+export function bindToNamespace<TClass extends { new (...args: any[]): any }>(
+  moduleClass: TClass,
+  that: InstanceType<TClass>
+): void {
+  // Bind `this` so namespaced is working correctly
+  for (const name of Object.getOwnPropertyNames(moduleClass.prototype) as Array<
+    keyof TClass | 'constructor'
+  >) {
+    if (name === 'constructor' || typeof that[name] !== 'function') {
```

Comment:
I would flip this, because `'typeof constructor === 'function'` so there is no need to explicitly check for the name, if the type doesn't match anyway.

```suggestion
    if (typeof that[name] !== 'function' || name === 'constructor') {
```